hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -1017,9 +1017,19 @@ private static Object rewriteRecordWithNewSchemaInternal(Object oldRecord, Schem
   public static String createFullName(Deque<String> fieldNames) {
     String result = "";
     if (!fieldNames.isEmpty()) {
-      List<String> parentNames = new ArrayList<>();
-      fieldNames.descendingIterator().forEachRemaining(parentNames::add);
-      result = parentNames.stream().collect(Collectors.joining("."));
+      Iterator<String> iter = fieldNames.descendingIterator();
+      result = iter.next();
+      if (!iter.hasNext()) {
+        return result;
+      }
+
+      StringBuilder sb = new StringBuilder();
+      sb.append(result);
+      while (iter.hasNext()) {
+        sb.append(".");
+        sb.append(iter.next());
+      }
+      result = sb.toString();
```

Comment:
Does this logic have a significant impact on performance? If not, the original implementation might offer better readability.

Reply:
> Does this logic have a significant impact on performance? If not, the original implementation might offer better readability.
![Screenshot from 2024-08-09 18-46-26](https://github.com/user-attachments/assets/3372c8cb-c524-402d-b810-8e200124a523)

Reply:
@wombatu-kun oh, my mistack, got it. And found rewrite record trigger it, thanks.